components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/APILoggerManager.java

```diff
@@ -0,0 +1,129 @@
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.apimgt.gateway;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.util.EntityUtils;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.apimgt.impl.dto.EventHubConfigurationDto;
+import org.wso2.carbon.apimgt.impl.utils.APIUtil;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class APILoggerManager {
+    private static final Log log = LogFactory.getLog(APILoggerManager.class);
+    private EventHubConfigurationDto getEventHubConfigurationDto;
+    public static final int retrievalRetries = 15;
+    public static final int retrievalTimeoutInSeconds = 15;
+    public static final String UTF8 = "UTF-8";
+
+    public Map<String, String> getPerAPILoggerList() throws APIManagementException{
+        Map<String, String> apiLoggerMap = new ConcurrentHashMap<>();
+        try {
+            String responseString = invokeService("/per-api-configs", MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+            JSONObject responseJson = new JSONObject(responseString);
+            JSONArray apiLogArray = responseJson.getJSONArray("apis");
+            for (int i = 0; i < apiLogArray.length(); i++){
+                JSONObject apiLoggerObject = apiLogArray.getJSONObject(i);
+                apiLoggerMap.put(apiLoggerObject.getString("context"), apiLoggerObject.getString("logLevel"));
+            }
+            if (log.isDebugEnabled()) {
+                log.debug("Response : " + responseString);
+            }
+        } catch (IOException ex) {
+            throw new APIManagementException("Error while calling internal service API", ex);
+        }
+        return apiLoggerMap;
+    }
+
+    public APILoggerManager() {
+        this.getEventHubConfigurationDto = org.wso2.carbon.apimgt.keymgt.internal.ServiceReferenceHolder.getInstance()
```

Comment:
Let's add proper formatting here

Reply:
You can refer https://tharika.medium.com/how-to-setup-code-formatting-in-my-ide-86374ec8fbd2